tests/lib/location/src/location_test.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <modem/at_monitor.h>
+#include <modem/location.h>
+#include <modem/lte_lc.h>
+#include <mock_nrf_modem_at.h>
+#include <mock_nrf_modem_gnss.h>
+#include <mock_modem_key_mgmt.h>
+#include <mock_rest_client.h>
+
+
+struct location_event_data test_location_event_data = {0};
+struct nrf_modem_gnss_pvt_data_frame test_pvt_data = {0};
+static bool location_callback_called_occurred;
+static bool location_callback_called_expected;
+
+K_SEM_DEFINE(event_handler_called_sem, 0, 1);
+
+/* at_monitor_dispatch() is implemented in at_monitor library and
+ * we'll call it directly to fake received AT commands/notifications
+ */
+extern void at_monitor_dispatch(const char *at_notif);
+/* method_gnss_event_handler() is implemented in the library and we'll call it directly
+ * to fake received GNSS event.
+ */
+extern void method_gnss_event_handler(int event);
+
+static void helper_location_data_clear(void)
+{
+	memset(&test_location_event_data, 0, sizeof(test_location_event_data));
+	memset(&test_pvt_data, 0, sizeof(test_pvt_data));
+}
+
+void setUp(void)
+{
+	helper_location_data_clear();
+	location_callback_called_occurred = false;
+	location_callback_called_expected = false;
+
```

Comment:
Would it be good to reset the `event_handler_called_sem` semaphore here as well?